src/cargo/sources/config.rs

```diff
@@ -125,6 +145,22 @@ impl<'gctx> SourceConfigMap<'gctx> {
         self.gctx
     }
 
+    fn load_and_overlay(
```

Comment:
nit: I feel like `and` in a name is a code smell, that you are making a fucntion do unrelated things.  Ideally, we find a name that gets at the heart of what distinguishes this from `load`

Reply:
What about `load_unreplaced`? Because the difference between `load` and `load_unreplaced` is that `load` does source replacement.

Reply:
Describing something by what it isn't doing is a bit messy.

Your name hints that `pub fn load` is `load_replaced`.  So maybe this could be `load_overlaid`.  Still not thrilled but probably the best we can do without blocking this further.